File: LXToolKitObjC/LXToolKitObjc/DJBusinessTools/Base/LXBaseVC.m
//
//  LXBaseVC.m
//  LXToolKitObjc
//
//  Created by lxthyme on 2022/6/22.
//
#import "LXBaseVC.h"
#import <Masonry/Masonry.h>
#import "DJLogger.h"

@interface LXBaseVC() {
}

@end

@implementation LXBaseVC
- (void)dealloc {
    commonlog(@"🛠DEALLOC[VC]: %@", NSStringFromClass([self class]));
}

#pragma mark -
#pragma mark - 🛠Life Cycle
- (void)viewDidLoad {
    [super viewDidLoad];
    // commonlog(@"🛠viewDidLoad: %@", NSStringFromClass([self class]));
    // Do any additional setup after loading the view.
    [self basePrepareUI];
}

#pragma mark -
#pragma mark - 📌UI Prepare & Masonry
- (void)basePrepareUI {
    [self baseMasonry];
}
#pragma mark getter / setter
#pragma mark Masonry
- (void)baseMasonry {
    NSString *clsName = NSStringFromClass([self class]);
    [self.view setMas_key:[NSString stringWithFormat:@"%@.view", clsName]];
    // MASAttachKeys(<#...#>)
}

#pragma mark Lazy Property
@end

File: LXToolKit/Classes/LXBaseVC.swift
//
//  LXBaseVC.swift
//  LXToolKit
//
//  Created by LXThyme Jason on 2019/11/12.
//

import UIKit

class LXBaseVC: UIViewController {
    // MARK: UI
    // MARK: Vaiables
    // MARK: Life Cycle
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
    }
    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)
    }
    override func viewWillDisappear(_ animated: Bool) {
        super.viewWillDisappear(animated)
    }
    override func viewDidDisappear(_ animated: Bool) {
        super.viewDidDisappear(animated)
    }
    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
        prepareUI()
    }

}

// MARK: LoadData
extension LXBaseVC {}

// MARK: Public Actions
extension LXBaseVC {}

// MARK: Private Actions
private extension LXBaseVC {}

// MARK: - UI Prepare & Masonry
private extension LXBaseVC {
    func prepareUI() {
        //[<#table#>].forEach(self.view.addSubview)
        masonry()
    }

    func masonry() {}
}
